samcli/lib/telemetry/metric.py

```diff
@@ -199,8 +200,11 @@ def wrapped(*args, **kwargs):
 
 def _get_project_details(hook_package_id: str, template_dict: Dict) -> ProjectDetails:
     if not hook_package_id:
-        project_type = ProjectTypes.CDK.value if is_cdk_project(template_dict) else ProjectTypes.CFN.value
-        return ProjectDetails(project_type=project_type, hook_package_id=None, hook_package_version=None)
+        hook_metadata = get_hook_metadata(template_dict)
+        if not hook_metadata:
+            project_type = ProjectTypes.CDK.value if is_cdk_project(template_dict) else ProjectTypes.CFN.value
+            return ProjectDetails(project_type=project_type, hook_package_id=None, hook_package_version=None)
+        hook_package_id = str(hook_metadata.get("HookPackageId"))
```

Comment:
only casting because mypy complained that `hook_package_id` was `str` but `hook_metadata.get("HookPackageId")` is `Any`. 

What should really happen if HookPackageId is None or not a string? Asking because it seems like even when we have an exception, we still return some project details (see a couple lines below, on the except).

Reply:
https://github.com/aws/aws-sam-cli/pull/4378/files#diff-848f38e3508120c82700f37e74bbd8688f7cbfb99dec7b44725771222f076fa6R212 

why are project_type and hook_package_id the same thing? I find this function signature weird.

Reply:
I am not sure if I correctly understood your concern, but `hook_package_id` and `project_type` can have different values. For example, once we start allowing others to add their own hooks, someone can add another Terraform hook, that can be named anything but the `project_type` will be always Terraform ...

Reply:
@sriram-mv  if you mean a few lines below when we catch InvalidHookPackageConfigException, I think it's because at that point it failed to read the config so we don't know the project type. So I'm guessing the next best thing is to use the package id? I don't even know if that's the behaviour we'd want but that's what I assumed it did right now.

Reply:
> So I'm guessing the next best thing is to use the package id

That's right. In theory it shouldn't ever enter that state, but if it does, then we can at least emit some metric instead of nothing.